docs/roles/integrator/integrating.md

```diff
@@ -0,0 +1,372 @@
+      "public_key": "ed25519:23vYngy8iL7q94jby3gszBnZ9JptpMf5Hgf7KVVa2yQi"
+    }
+  ]
+}
+
+# this example used HTTPie - a CLI, cURL-like tool for humans available at http://httpie.org
+```
+
+Discovering the details of a specific access key for a given account.
+
+```sh
+PUBLIC_KEY=ed25519:23vYngy8iL7q94jby3gszBnZ9JptpMf5Hgf7KVVa2yQi
+
+# query format: access_key/<account_id>/<access_key_public_part>
+http post http://rpc.nearprotocol.com jsonrpc=2.0 method=query  \
+                                      params:="[\"access_key/test.near/$PUBLIC_KEY\",\"\"]" \
+                                      id="placeholder"
+
+# the request above returns details about an access key for given account with this public key.
+# If there is no such access key, it returns nothing.
+# ------------------------------------------------------------------------------
+# response
+{
+  "id": "placeholder",
+  "jsonrpc": "2.0",
+  "result": {
+    "nonce": 1,
+    "permission": "FullAccess"
+  }
+}
+
+# this example used HTTPie - a CLI, cURL-like tool for humans available at http://httpie.org
+```
+
+### Account Events
+
+The NEAR platform emits events related to changes on the blockchain.  Developers can subscribe to these events and receive changes as blocks are processed.  Events are organized around account compute and storage.
+
+<blockquote class="warning">
+<strong>work in progress</strong> <span>support for events is currently under heavy development</span><br><br>
+
+Discussion is here: https://github.com/nearprotocol/nearcore/issues/1546
+
+Briefly, as of Nov 2019, the Near team is developing a [WAMP-proto](https://wamp-proto.org/_static/gen/wamp_latest.html#protocol-overview) router protocol with support for a [dealer [RPC] role](https://wamp-proto.org/_static/gen/wamp_latest.html#remote-procedure-calls) with an advanced extension for [progressive calls](https://wamp-proto.org/_static/gen/wamp_latest.html#progressive-call-results) using Actix WebSocket.
+
+</blockquote>
+
+### Create key pair (offline)
+
+NEAR supports generating public/private key pairs from a random seed or using a string to generate a public key from a given private key.
+
+Generating key pairs two ways and comparing them
+
+```js
+// generating a key pair
+const keyPair = nearlib.utils.key_pair.KeyPairEd25519.fromRandom();
+
+// generating another key pair based on the private key of the key pair above
+const newKeyPair = nearlib.utils.key_pair.KeyPair.fromString(keyPair.toString());
+
+// the two will match
+console.assert(newKeyPair.secretKey === keyPair.secretKey);
+```
+
+Generating a key pair to sign and verify a message
+
+```js
+// generate a key pair
+const keyPair = nearlib.utils.key_pair.KeyPairEd25519.fromRandom();
+
+// create and sign a message
+const message = new Uint8Array(sha256.array('message'));
+const signature = keyPair.sign(message);
+
+// the message is verifiably signed by the correct key
+console.assert(keyPair.verify(message, signature.signature));
+```
```

Comment:
we probably should add another section here on how to create account. You have it in the example below

Reply:
will update

Reply:
⏳